### ISSUE TITLE
Registration within selected ROIs

### ISSUE THREAD

**Issue by anranli0** [MathOnco/valis#27]
I have pairs of slides that contain artifacts / irrelevant information and ROIs for each slide (annotated in QuPath and stored as coordinates). I would like to register only the selected regions on a given pair of slides. Does valis offer such option? If not, what modifications are needed in order to achieve that?

Any advice would be appreciated. Thank you!

**Comment by cdgatenbee** [comment]
Hi @anranli0 ,
Apologies for the delayed response. At the moment, there isn't exactly a built in way to to register only ROIs, but one option could be to use valis to slice out each ROI (using the annotated coordinates), saving the images in a single folder, e.g. roi1 contains roi1_img1, roi1_img2, roi2 contains roi2_img1, roi2_img2, etc... and using valis to only register the images in each roi folder. If the regions are small they could be saved as pngs, but if they're large you could used `slide_io.save_ome_tiff` to save them as compressed ome.tiffs. Do you think something like that would work in this case? Another option would be to use the ROI to mask the images, and perform registration on the masked images. This would be a little more complicated but if you think it would be a better solution in your case I'd be happy to help put something together. 

Best,
Chandler

**Comment by anranli0** [comment]
Hi @cdgatenbee ,

Thank you for your response! 

I ended up going for option 2. Here's what I did:
- Save the polygon vertices of each ROI as a separate file
- Add an attribute to Slide objects to associate an ROI file with each slide
- Convert ROI to mask and take an intersection with the mask generated by valis
- perform registration

It worked for my purpose but I did have to run registration and save a set of ome.tiff slides for each ROI. I totally missed the possibility of saving only regions of a slide and might try option 1 or saving a sliced out region of a registered slide for memory efficiency.

Thank you for your help!
